text/0000-sawtooth-governance.md

```diff
@@ -0,0 +1,348 @@
+broader the consensus, the better. But at the very least, **consensus within the
+members of the subteam should be the norm for most decisions.** If the root team
+has done its job of communicating the values and priorities, it should be
+possible to fit the debate about the RFC into that framework and reach a fairly
+clear outcome.
+
+[consensus]: http://en.wikipedia.org/wiki/Consensus_decision-making
+
+#### Lack of consensus
+
+In some cases, though, consensus cannot be reached. These cases tend to split
+into two very different camps:
+
+* "Trivial" reasons, e.g., there is not widespread agreement about naming, but
+  there is consensus about the substance.
+
+* "Deep" reasons, e.g., the design fundamentally improves one set of concerns
+  at the expense of another, and people on both sides feel strongly about it.
+
+In either case, an alternative form of decision-making is needed.
+
+* For the "trivial" case, usually either the RFC shepherd or subteam leader
+  will make an executive decision.
+
+* For the "deep" case, the subteam leader is empowered to make a final
+  decision, but should consult with the rest of the root team before doing so.
+
+#### Final Comment Period (FCP)
+
+Each RFC has a shepherd drawn from the relevant subteam. The shepherd is
+responsible for driving the consensus process -- working with both the RFC
+author and the broader community to dig out problems, alternatives, and
+improved design, always working to reach broader consensus.
+
+At some point, the RFC comments will reach a kind of "steady state", where no
+new tradeoffs are being discovered, and either objections have been addressed,
+or it's clear that the design has fundamental downsides that need to be
+weighed.
+
+At that point, the shepherd will announce that the RFC is in a "final comment
+period" (which lasts for one week). This is a kind of "last call" for strong
+objections to the RFC. **The announcement of the final comment period for an
+RFC should be very visible**; it should be included in the subteam's periodic
+communications.
+
+After the final comment period, the subteam can make a decision on the RFC. The
+role of the subteam at that point is *not* to reveal any new technical issues
+or arguments; if these come up during discussion, they should be added as
+comments to the RFC, and it should undergo another final comment period.
+
+Instead, the subteam decision is based on **weighing the already-revealed
+tradeoffs against the project's priorities and values** (which the root team is
+responsible for setting, globally). In the end, these decisions are about how
+to weight tradeoffs. The decision should be communicated in these terms,
+pointing out the tradeoffs that were raised and explaining how they were
+weighted, and **never introducing new arguments**.
+
+### Initial Subteams
+
+The following is a proposed list of initial subteams which is subject to change
+at the discretion of the root team, once it has formed.
```

Comment:
I would prefer if the sub-teams are focussed on software components that have their own separate existence, such as Seth and the Go SDK, where there is a definite releasable, versioned artifact they are responsible for. Other concerns like community outreach should be handled by the each team, taking their lead/direction from core.

Reply:
None of the Sawtooth components have a separate existence per-se, as it is one big project. We have different components but not different projects. Conceptually, something like Seth could be elevated to a proper Hyperledger project itself at some point.

Releases, publishing of artifacts, etc. need to be coordinated across the components. We could release components on their own release schedules (this makes a lot of sense for components which are not tightly coupled), but general strategy should roughly be the same (same tagging/branching approach, etc.).

For the SDKs, the sub-team will be represented by maintainers of all the SDKs (ideally) as it is important that we have similar features and design across the SDKs.